src/CascadeMinimizer.cc

```diff
@@ -171,11 +194,25 @@ bool CascadeMinimizer::minos(const RooArgSet & params , int verbose ) {
       if (simnll) simnll->clearZeroPoint();
    }
 
+   if (simnllbb && runtimedef::get("MINIMIZER_analytic")) {
+     simnllbb->setAnalyticBarlowBeeston(false);
+     // minimizer_.reset(new RooMinimizerOpt(nll_));
+   }
+
    return (iret != 1) ? true : false; 
 }
 
 bool CascadeMinimizer::hesse(int verbose ) {
    
+   cacheutils::CachingSimNLL *simnllbb = dynamic_cast<cacheutils::CachingSimNLL *>(&nll_);
```

Comment:
In general assume that when we run hesse we want all the analytic bin-by-bin parameters included in the covariance matrix. We therefore have to reset the minimizer again (to account for bbb's going from const -> non-const) and minimize again so that hesse will actually run.